internal/cli/universal_login_templates.go

```diff
@@ -0,0 +1,452 @@
+	)
+
+	//go:embed data/branding/storybook/*
+	templatePreviewAssets embed.FS
+
+	//go:embed data/branding/tenant-data.js
+	tenantDataAsset string
+
+	//go:embed data/branding/default-template.liquid
+	templateBasic string
+
+	//go:embed data/branding/image-template.liquid
+	templateWithImage string
+
+	//go:embed data/branding/footer-template.liquid
+	templateWithFooter string
+
+	templateBody = Flag{
+		Name:       "Template",
+		LongForm:   "template",
+		ShortForm:  "t",
+		Help:       "Custom page template for Universal Login.",
+		IsRequired: true,
+	}
+
+	templateOptions = pickerOptions{
+		{"Basic template", templateBasic},
+		{"Template with login box and background image", templateWithImage},
+		{"Template with page footers", templateWithFooter},
+	}
+)
+
+// TemplateData contains all the variables we project onto our embedded go
+// template. These variables largely resemble the same ones in the auth0
+// branding template.
+type TemplateData struct {
+	Filename        string
+	Clients         []ClientData
+	PrimaryColor    string
+	BackgroundColor string
+	LogoURL         string
+	TenantName      string
+	Body            string
+}
+
+// ClientData is a minimal representation of an Auth0 Client as defined in the
+// management API. This is used within the branding machinery to populate the
+// tenant data.
+type ClientData struct {
+	ID      string `json:"id"`
+	Name    string `json:"name"`
+	LogoURL string `json:"logo_url,omitempty"`
+}
+
+func showBrandingTemplateCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "show",
+		Args:  cobra.NoArgs,
+		Short: "Display the custom template for Universal Login",
+		Long:  "Display the custom template for the Universal Login experience.",
+		Example: `  auth0 universal-login templates show
+  auth0 ul templates show`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var currentTemplate *management.BrandingUniversalLogin
+			if err := ansi.Waiting(func() (err error) {
+				currentTemplate, err = cli.api.Branding.UniversalLogin()
+				if err != nil {
+					if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+						return nil
+					}
+					return err
+				}
+
+				return nil
+			}); err != nil {
+				return fmt.Errorf("failed to load the Universal Login template: %w", err)
+			}
+
+			cli.renderer.Heading("universal login template")
+
+			if currentTemplate == nil {
+				cli.renderer.Infof("No custom template found. To set one, run: `auth0 universal-login templates update`.")
+			}
+
+			fmt.Println(currentTemplate.GetBody())
+
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func updateBrandingTemplateCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update",
+		Args:  cobra.NoArgs,
+		Short: "Update the custom template for Universal Login",
+		Long:  "Update the custom template for the New Universal Login Experience.",
+		Example: `  auth0 universal-login templates update
+  auth0 ul templates update`,
```

Comment:
Removed support for piping the template as an input to the command as that would skip the preview mechanism due to the "shouldPrompt" logic we have and it would just make things more confusing.